hybrid/decomposers.py

```diff
@@ -725,31 +729,34 @@ def next(self, state, **runopts):
         return state.updated(subproblem=subbqm)
 
 
-def _all_minimal_covers(edgelist):
-    """Create all minimal covers, provided graphs contain fewer than 17 edges."""
-
-    if len(edgelist) > 16:
-        raise ValueError(
-            f'len(edgelist) = {len(edgelist)} > 16. Exponential scaling of this method means it '
-            'should only be used for small graphs. The use case for this function '
-            'is in finding coverings over unyielded lattice subgraphs, which in '
-            'the current online generation of processors are sufficiently small '
-            'so as to allow a brute force method.')
-
-    minimum_coverings = []
-    for cover in map(set, product(*edgelist)):
-        newmin = []
-        for s in minimum_coverings:
-            if s < cover:
-                break
-            elif cover < s:
-                pass
-            else:
-                newmin.append(s)
+def _good_cover(edgelist, brute_force_threshold=16):
+    # Attempt to solve by brute force (tree decomposition, under tree decomposition)
+    try:
+        tds = TreeDecompositionSolver()
+        G = nx.from_edgelist(edgelist)
+        coverLTW = dnx.algorithms.cover.min_vertex_cover(G=G, sampler=tds)
+        return coverLTW
+    except:
```

Comment:
You might want to be more specific here with which exception is expected. Also, to @randomir's point below, unless `min_vertex_cover` or `min_weighted_vertex_cover` mutate `G`, its creation could be pulled outside the try-except.

Reply:
Yes, this was lazy of me. It is possible to anticipate the algorithm failure and bypass the exception. (implemented)